Bsii.Dotnet.Utils/Reflection/ObjectFlattener.cs

```diff
@@ -208,5 +209,53 @@ public static bool IsTerminalType(Type t)
                    || t == typeof(DateTime)
                    || t == typeof(DateTimeOffset);
         }
+        /// <summary>
+        /// Flattens a dictionary to a dictionary of<key,<parameterName,parameterValue>>
+        /// </summary>
+        /// <typeparam name="TKey"></typeparam>
+        /// <typeparam name="TValue"></typeparam>
+        /// <param name="dict">the given dictionary to flatten</param>
+        /// <param name="delimiter">delimiter string to input between parameters</param>
+        /// <param name="shortNames">option to choose only the string after the last delimiter or show full string</param>
+        /// <returns></returns>
+        public static Dictionary<TKey, Dictionary<string, string>> FlattenDictionary<TKey, TValue>(Dictionary<TKey, TValue> dict, string delimiter = ".", bool shortNames = false)
+        {
+            var dic = new Dictionary<TKey, Dictionary<string, string>>();
+            foreach (var kvp in dict)
+            {
+                var hidRes = new Dictionary<string, string>();
+                FlattenObjectToDictionary(kvp.Value, "", hidRes, delimiter, shortNames);
+                dic.Add(kvp.Key, hidRes);
+            }
+            return dic;
+        }
+
+        private static void FlattenObjectToDictionary(object obj, string prefix, Dictionary<string, string> dic, string delimiter, bool shortNames)
+        {
+            if (obj == null) return;
+            Type objType = obj.GetType();
+            PropertyInfo[] properties = objType.GetProperties();
+            var accessor = TypeAccessor.Create(objType);
+            foreach (var property in properties)
+            {
+                object propValue = accessor[obj, property.Name];
+                if (propValue == null)
+                {
+                    return;
+                }
+                var newPrefix = shortNames ? "" : prefix.Length == 0 ? prefix : $"{prefix}{delimiter}";
+                if (property.PropertyType.Assembly == objType.Assembly && !property.PropertyType.IsEnum)
```

Comment:
I still thinks this line is very impicit - should be something like 
```suggestion
                if (!(property.PropertyType.IsPrimitvie || 
                property.PropertyType.IsDateTime||
               property.PropertyType.IsEnum))
```
something like that

Reply:
we tried this and it didn't work as expected

Reply:
it was quick debugging... I think this can be resolved

Reply:
no option to check DateTime